graphviz/files.py

```diff
@@ -170,13 +171,14 @@ def render(self, filename=None, directory=None, view=False, cleanup=False):
             subprocess.CalledProcessError: If the exit status is non-zero.
             RuntimeError: If viewer opening is requested but not supported.
         """
-        filepath = self.save(filename, directory)
+        if cleanup:
+            filepath = self.save(tempfile.mkstemp('.gv')[1])
```

Comment:
after looking at the code more this makes total sense. seems like it involves adding a `cleanup` arg to `save` though and modifying the test from `save.assert_called_once_with(None, None)` to save.assert_called_once_with(None, None, True)`